lib/parse/parser_interpreter.es6.js

```diff
@@ -0,0 +1,299 @@
+          name: 'value',
+        },
+      ],
+    },
+    {
+      name: 'interpretKeyword',
+      returns: 'foam.mlang.predicate.AbstractPredicate',
+      args: [
+        {
+          typeName: 'String',
+          documentation: 'The keyword in a keyword query fragment.',
+          name: 'kw',
+        },
+      ],
+    },
+  ],
+});
+
+foam.CLASS({
+  name: 'DefaultQueryInterpreter',
+  package: 'org.chromium.parse',
+  implements: [
+    'foam.mlang.Expressions',
+    'org.chromium.parse.QueryInterpreter',
+  ],
+
+  documentation: `Basic interpreter that matches key/values to properties and
+      interprets keywords in the natural way.`,
+
+  requires: ['foam.core.Property'],
+
+  methods: [
+    function interpretKeyValue(key, value, parser) {
+      const property = parser.of.getAxiomByName(key);
+      if (!property || !this.Property.isInstance(property)) {
+        return this.delegate ?
+            this.delegate.interpretKeyValue(key, value, parser) :
+            this.interpretKeyword(`${key}:${value}`, parser);
+      }
+      return this.EQ(property, value);
+    },
+    function interpretKeyword(kw, parser) {
+      return this.KEYWORD(kw);
+    },
+  ],
+});
+
+foam.CLASS({
+  name: 'ReleaseApiQueryInterpreter',
+  package: 'org.chromium.parse',
+  implements: [
+    'foam.mlang.Expressions',
+    'org.chromium.mlang.GridExpressions',
+    'org.chromium.mlang.ParseExpressions',
+    'org.chromium.parse.QueryInterpreter',
+  ],
+
+  documentation: `Interpreter for handling release key matching and
+      count-of-selected-releases. E.g.,
+
+      count:3 => "Exactly 3 selected releases support API"
+
+      fir59win10:0 => [prefix match alphabetical and numeric groups]
+                   => "Not shipped (last 0 is false) in Firefox 59 Windows 10"
+
+      chr60:t => [prefix match alphabetical and numeric groups]
+              => "Shipped (t for false) in Chrome 60 on all known platforms"`,
+
+  requires: [
+    'org.chromium.apis.web.CompatProperty',
+    'org.chromium.parse.DefaultQueryInterpreter',
+  ],
+  imports: [
+    'selectable',
+    'selected',
+  ],
+
+  properties: [
+    {
+      name: 'delegate',
+      factory: function() {
+        return this.DefaultQueryInterpreter.create();
+      },
+    },
+  ],
+
+  methods: [
+    function interpretKeyValue(key, value, parser) {
+      // When delegate is set and interpretation is null, pass to delegate for
+      // interpretation.
+      const ret = this.delegate ?
+          (v => v === null ?
+              this.delegate.interpretKeyValue(key, value, parser) : v) :
+          (v => v === null ?
+              this.interpretKeyword(`${key}:${value}`, parser) : v);
+
+      if (key === 'count') {
+        return ret(this.interpretCount(value, parser));
+      }
+      const columns = this.getMatchingColumns(key);
+      if (columns.length === 0) return ret(null);
+      // Starts with 't' or non-zero digit => true;
```

Comment:
This feels unnecessarily complicated to me, but I have no context for why it's this way.

Reply:
This is just my wild guess as to how programmers might try to use a Boolean value label. `t`, `true`, `TRUE`, `truthy`, and `1` (as opposed to `0`) seemed like reasonable "guesses at how to construct a valid query". I'm open to feedback on why this might be confusing or dangerous.

Reply:
They are already learning a new query language. Telling them what you want them to enter seems like a better approach than letting them put in random garbage with little to no guidance on what that garbage is going to do.